src/runtime_src/core/pcie/tools/xbutil/xbutil.h

```diff
@@ -464,17 +489,32 @@ class device {
         }
 
         //Soft kernel info below
+        std::vector<ps_kernel_data> psKernels;
+        if (getPSKernels(psKernels) < 0) {
+            std::cout << "WARNING: 'ps_kernel' invalid. Has the PS kernel been loaded? See 'xbutil program'.\n";
+            return 0;
+        }
+
+        int psk_inst = 0;
+        uint32_t num_scu = 0;
         for (unsigned int i = computeUnits.size(); i < parseComputeUnitNum(custat); i++) {
             uint32_t status = parseComputeUnitStat(custat, i, cu_stat::stat);
             uint32_t usage = parseComputeUnitStat(custat, i, cu_stat::usage);
-	    auto name = parseComputeUnitName(custat, i);
+            std::string name = psKernels.at(psk_inst).pkd_sym_name;
```

Comment:
Please add a comment explaining what is happening here, preferably with an example